src/socks5proxy/src/socks5.h

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SOCKS5_H
+#define SOCKS5_H
+
+#include <QObject>
+#include <QTcpServer>
+
+class Socks5 final : public QObject {
+  Q_OBJECT
+  Q_PROPERTY(uint16_t connections READ connections NOTIFY connectionsChanged);
+
+ public:
+  explicit Socks5(QObject* parent, uint16_t port, QHostAddress listenAddress);
```

Comment:
Minor nit: Qt convention usually puts the `QObject* parent` as the last parameter and gives it a default value of `nullptr` This allows the caller to omit it, in which case it is bound to the lifetime of the `qApp`.